tests/marine/benchmark/main.py

```diff
@@ -0,0 +1,190 @@
+        )
+        assert passed
+        assert result is None
+
+
+@benchmark_wrapper
+def benchmark_3_fields(packets: List[BenchmarkPacket]):
+    for packet in packets:
+        fields_to_extract = packet.fields_to_extract[:3]
+        expected = dict()
+        for field in fields_to_extract:  # TODO improve this
+            expected[field] = packet.expected_parse_result[field]
+
+        passed, result = marine_instance.filter_and_parse(
+            packet.packet, fields=packet.fields_to_extract[:3]
+        )
+        assert passed
+        assert expected == result
+
+
+@benchmark_wrapper
+def benchmark_8_fields(packets: List[BenchmarkPacket]):
+    for packet in packets:
+        passed, result = marine_instance.filter_and_parse(
+            packet.packet, fields=packet.fields_to_extract
+        )
+        assert passed
+        assert packet.expected_parse_result == result
+
+
+@benchmark_wrapper
+def benchmark_bpf_and_display_filter_and_3_fields(packets: List[BenchmarkPacket]):
+    for packet in packets:
+        fields_to_extract = packet.fields_to_extract[:3]
+        expected = dict()
+        for field in fields_to_extract:  # TODO improve this
+            expected[field] = packet.expected_parse_result[field]
+        passed, result = marine_instance.filter_and_parse(
+            packet.packet,
+            bpf=packet.good_bpf,
+            display_filter=packet.good_display_filter,
+            fields=packet.fields_to_extract[:3],
+        )
+        assert passed
+        assert expected == result
+
+
+@benchmark_wrapper
+def benchmark_bpf_and_display_filter_and_8_fields(packets: List[BenchmarkPacket]):
+    for packet in packets:
+        passed, result = marine_instance.filter_and_parse(
+            packet.packet,
+            bpf=packet.good_bpf,
+            display_filter=packet.good_display_filter,
+            fields=packet.fields_to_extract,
+        )
+        assert passed
+        assert packet.expected_parse_result == result
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--benchmark", choices=["all"] + list(benchmark_functions.keys())
+    )
+    args = parser.parse_args()
+    generated_packets = generate_packets(
+        AUTO_RESET_COUNT
+    )  # TODO: this might need to be AUTO_RESET_COUNT + 1
+    benchmark_start_used_memory = get_used_memory_in_mb()
+
+    if not args.benchmark or args.benchmark == "all":
+        # TODO: I can take these from benchmark_functions, but I want them executed in this order
+        benchmark_bpf(generated_packets)
+        benchmark_display_filter(generated_packets)
+        benchmark_bpf_and_display_filter(generated_packets)
+        benchmark_3_fields(generated_packets)
+        benchmark_8_fields(generated_packets)
+        benchmark_bpf_and_display_filter_and_3_fields(generated_packets)
+        benchmark_bpf_and_display_filter_and_8_fields(generated_packets)
+    elif args.benchmark in benchmark_functions:
```

Comment:
It would be cool if this option got a glob pattern and not just an exact name (but that's obviously an overkill)

Reply:
I'll add a todo :)